src/driver/amdxdna/aie2_pci.c

```diff
@@ -1044,6 +1044,36 @@ static int aie2_get_force_preempt_state(struct amdxdna_client *client,
 	return 0;
 }
 
+static int aie2_get_resource_info(struct amdxdna_client *client,
+				  struct amdxdna_drm_get_info *args)
+{
+	struct amdxdna_drm_get_resource_info res_info = {};
+	struct amdxdna_dev *xdna = client->xdna;
+	struct amdxdna_dev_hdl *ndev = xdna->dev_handle;
+	const struct amdxdna_dev_priv *pdev = ndev->priv;
+	int ret = 0;
+
+	res_info.ipu_clk_max = pdev->dpm_clk_tbl[ndev->max_dpm_level - 2].hclk;
+	res_info.ipu_task_max = pdev->hwctx_limit;
+	res_info.ipu_task_curr = ndev->hwctx_cnt;
+	u32 total_col = ndev->total_col;
+	u32 hclk_freq = ndev->hclk_freq;
```

Comment:
Do NOT declare local variable in the middle of the code. Unless it is a new block. I know this is suggested in C++. But this not the case in Linux.